inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -427,3 +426,56 @@ def normalize_journal_titles(obj, eng):
 
             if result:
                 obj.data['publication_info'][index]['journal_record'] = result.records_metadata_json['self']
+
+
+@with_debug_logging
+def set_refereed_and_fix_document_type(obj, eng):
+    """Set the ``refereed`` field using the Journals DB.
+
+    Searches in the Journals DB if the current article was published in journals
+    that we know for sure to be peer-reviewed, or that publish both peer-reviewed
+    and non peer-reviewed content but for which we can infer that it belongs to
+    the former category, and sets the ``refereed`` key in ``data`` to ``True`` if
+    that was the case. If instead we know for sure that all journals in which it
+    published are **not** peer-reviewed we set it to ``False``.
+
+    Also replaces the ``article`` document type with ``conference paper`` if the
+    paper was only published in non refereed proceedings.
+
+    Args:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        None
+
+    """
+    journals = replace_refs(get_value(obj.data, 'publication_info.journal_record'), 'db')
+    if not journals:
+        return
+
+    is_published_in_a_refereed_journal = any(
```

Comment:
@jacquerie `is_published_in_a_refereed_journal` should be called `is_published_in_a_refereed_journal_that_does_not_publish_proceedings` in order to be accurate.